src/codegen/doc/format.rs

```diff
@@ -54,121 +55,207 @@ fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
+                    find_constant_or_variant(symbol_name, env)
+                } else {
+                    let method_name = caps.get(3).map(|m| m.as_str().trim_start_matches('.'));
+                    // would be #
+                    find_type_by_name(symbol_name, method_name, env)
                 }
             }
-        } else {
-            format!("{}`{}{}`", &caps[1], &caps[3], member)
         }
     });
-    let out = GDK_GTK.replace_all(&out, |caps: &Captures<'_>| {
-        format!("`{}`", lookup(&caps[0]))
-    });
+    let out = GDK_GTK.replace_all(&out, |caps: &Captures<'_>| find_type(&caps[2], env));
     let out = TAGS.replace_all(&out, "`$0`");
     SPACES.replace_all(&out, " ").into_owned()
 }
+
+fn find_type_by_name(symbol: &str, method_name: Option<&str>, env: &Env) -> String {
+    let symbols = env.symbols.borrow();
+    if let Some(method) = method_name {
+        if let Some((obj_info, fn_info)) = env.analysis.find_object_by_function(
+            env,
+            |o| o.full_name == symbol,
+            |f| f.name == method,
+        ) {
+            let sym = symbols.by_tid(obj_info.type_id).unwrap(); // we are sure the object exists
+            let (type_name, visible_type_name) = obj_info.generate_doc_link_info(fn_info);
+            let name = sym.full_rust_name().replace(&obj_info.name, &type_name);
+
+            fn_info.doc_link(Some(&name), Some(&visible_type_name))
+        } else if let Some((record_info, fn_info)) = env.analysis.find_record_by_function(
+            env,
+            |r| r.type_(&env.library).c_type == symbol,
+            |f| f.name == method,
+        ) {
+            let sym_name = symbols
+                .by_tid(record_info.type_id)
+                .unwrap()
+                .full_rust_name(); // we are sure the object exists
+            fn_info.doc_link(Some(&sym_name), None)
+        } else {
+            format!("`{}::{}`", symbol, method)
+        }
+    } else {
+        find_type(symbol, env)
+    }
+}
+
+fn find_constant_or_variant(symbol: &str, env: &Env) -> String {
+    let symbols = env.symbols.borrow();
+    if let Some(const_info) = env
+        .analysis
+        .constants
+        .iter()
+        .find(|c| c.glib_name == symbol)
+    {
+        // for whatever reason constants are not part of the symbols list
+        format!("[`{name}`][crate::{name}]", name = const_info.name)
+    } else if let Some((flag_info, member_info)) = env.analysis.flags.iter().find_map(|f| {
+        f.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol)
+            .map(|m| (f, m))
+    }) {
+        let sym = symbols.by_tid(flag_info.type_id).unwrap();
+        format!(
+            "[`{flag_name}::{member_name}`][crate::{parent}::{member_name}]",
+            member_name = nameutil::bitfield_member_name(&member_info.name),
+            flag_name = flag_info.name,
+            parent = sym.full_rust_name()
+        )
```

Comment:
Added an `assert_eq!(flag_info.name, sym.full_rust_name())` here, and it doesn't trigger on any of GStreamer. I don't think we need this extra separation, unless you have found a case where the Rust type linked to should be different from the name of the flag? IMO I'd like to see the namespace (that `full_rust_name()` provides) in that case.